src/acceptance-test/java/org/zalando/nakadi/service/job/ExclusiveJobWrapperAT.java

```diff
@@ -0,0 +1,99 @@
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+import org.zalando.nakadi.webservice.BaseAT;
+import org.zalando.nakadi.webservice.utils.ZookeeperTestUtils;
+
+import java.nio.charset.Charset;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.lessThan;
+import static org.mockito.Mockito.when;
+import static org.zalando.nakadi.utils.TestUtils.randomUUID;
+
+public class ExclusiveJobWrapperAT extends BaseAT {
+
+    private static final CuratorFramework CURATOR = ZookeeperTestUtils.createCurator(ZOOKEEPER_URL);
+    private static final int TEST_JOB_PERIOD_MS = 3 * 60 * 60 * 1000; // 3 hours
+
+    private ObjectMapper objectMapper;
+    private final ZooKeeperHolder zkHolder;
+    private AtomicBoolean jobExecuted;
+    private Runnable dummyJob;
+    private ExclusiveJobWrapper jobWrapper;
+    private String lockPath;
+    private String latestPath;
+
+    public ExclusiveJobWrapperAT() {
+        objectMapper = new JsonConfig().jacksonObjectMapper();
+        zkHolder = Mockito.mock(ZooKeeperHolder.class);
+        when(zkHolder.get()).thenReturn(CURATOR);
+    }
+
+    @Before
+    public void before() throws Exception {
+        final String testJobName = randomUUID();
+
+        lockPath = ZKPaths.makePath(ExclusiveJobWrapper.NAKADI_JOBS_PATH, testJobName, "lock");
+        latestPath = ZKPaths.makePath(ExclusiveJobWrapper.NAKADI_JOBS_PATH, testJobName, "latest");
+
+        jobExecuted = new AtomicBoolean(false);
+        dummyJob = () -> jobExecuted.set(true);
+
+        jobWrapper = new ExclusiveJobWrapper(zkHolder, objectMapper, testJobName, TEST_JOB_PERIOD_MS);
+    }
+
+    @Test
+    public void whenItsNotTimeToRunThenJobIsNotExecuted() throws Exception {
+        // set latest job execution as two hours ago
+        createLatestNode(2);
+
+        jobWrapper.runJobLocked(dummyJob);
+        assertThat(jobExecuted.get(), is(false));
+    }
+
+    @Test
+    public void whenJobIsAlreadyRunningOnAnotherNodeThenJobIsNotExecuted() throws Exception {
+        // set latest job execution as 30 hours ago
+        createLatestNode(30);
+        // another Nakadi instance already performs cleaning
+        CURATOR.create().creatingParentsIfNeeded().forPath(lockPath);
+
+        jobWrapper.runJobLocked(dummyJob);
+        assertThat(jobExecuted.get(), is(false));
+    }
+
+    @Test
+    public void whenExecuteJobThenOk() throws Exception {
+        jobWrapper.runJobLocked(dummyJob);
+        assertThat(jobExecuted.get(), is(true));
+
+        assertThat("lock node should be removed", CURATOR.checkExists().forPath(lockPath), is(nullValue()));
+
+        final byte[] data = CURATOR.getData().forPath(latestPath);
+        final DateTime lastCleaned = objectMapper.readValue(new String(data, Charsets.UTF_8), DateTime.class);
```

Comment:
I think the current "standard" is to use guava Charsets class.
Changed https://github.com/zalando/nakadi/pull/611/commits/c869f991250d95cd892d4e79300f3ca724b9466d